### PR TITLE
[Enhancement] Column prune for olap scan node

### PR DESCRIPTION
**Pull request by Kikyou1997**
# Proposed changes

Add Column prune for olap scan node based on the project planner framework

## Problem Summary:

Implement the `initOutputSlotIds`  method

## Checklist(Required)

1. Does it affect the original behavior: (Yes)
2. Has unit tests been added: (Yes)
3. Has document been added or modified: (No Need)
4. Does it need to update dependencies: (No)
5. Are there any changes that cannot be rolled back: (No)

### CODE REVIEW

**Comment by yiguolei** [comment]
Do not need any changes in BE?

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java [hunk 1]
```diff
@@ -941,4 +943,17 @@ public DataPartition constructInputPartitionByDistributionInfo() throws UserExce
             return DataPartition.RANDOM;
         }
     }
+
+    @Override
+    public void initOutputSlotIds(Set<SlotId> requiredSlotIdSet, Analyzer analyzer) throws NotImplementedException {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:948]: Although fe calculates the output slot of the scan node, if be does not make any changes, the function of project cannot be realized.

File: fe/fe-core/src/main/java/org/apache/doris/planner/PlanNode.java [hunk 2]
```diff
@@ -984,4 +984,17 @@ public String toString() {
         sb.append("\n").append(getNodeExplainString("", TExplainLevel.BRIEF));
         return sb.toString();
     }
+
+    /**
+     * Used to append some common explains to output
+     */
+    protected void appendCommonExplainString(String detailPrefix, StringBuilder output) {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/planner/PlanNode.java:991]: It seems that there is already a getExplainString in the plan node for common string output.

**Review by EmmyMiao87 — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java (hunk 1 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/java/org/apache/doris/planner/OlapScanNode.java:948]: OK, I'll add supports in BE later

**Comment by zenoyang** [comment]
Excuse me, why close pr, is there any plan for this?

**Pull request closed without merging**